### PR TITLE
fix(prompt-adapter): resolve Llama4 tool calling 500 errors

### PR DESCRIPTION
**Pull request by skamenan7**
## Problem

Llama4 tool calling was causing 500 errors due to missing `arguments_json` parameter in ToolCall objects. Systems downstream expect this field to be present for proper tool execution.

## Solution

- **Fixed missing `arguments_json` parameter** in `decode_assistant_message_from_content` method
- **Added comprehensive test suite** with 5 pytest-compatible tests covering:
  - Custom tool calls  
  - Builtin tool calls (brave_search, code_interpreter, etc.)
  - Complex argument structures
  - Edge cases and validation

## Testing

```bash
pytest models/llama4/tests/api/test_chat_format.py -v
======= 5 passed in 1.46s =======
```

## Related

- Addresses Issue #2584 (PR #2663 reviewer feedback)
- Fixes compatibility with systems expecting `arguments_json` field
- Maintains backward compatibility with existing `arguments` field

## Validation

All tool call types now properly include `arguments_json` field with valid JSON serialization that matches the `arguments` dict.

### CODE REVIEW

**Comment by skamenan7** [comment]
Hi @leseb , forgot to comment here that it is ready for review

**Review by ashwinb — APPROVED** [review]
looks good

File: models/llama4/tests/api/test_chat_format.py [hunk 1]
```diff
@@ -0,0 +1,191 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the terms described in the LICENSE file in
+# top-level folder for each specific model found within the models/ directory at
+# the top-level of this source tree.
+
+import json
+import unittest
```
Comment (leseb) [inline models/llama4/tests/api/test_chat_format.py:9]: please use pytest
